src/main/jhipster/src/main/webapp/app/germplasm-details/pedigree/germplasm-neighborhood-tree.component.ts

```diff
@@ -0,0 +1,95 @@
+import { Component, Input, OnInit, ViewEncapsulation } from '@angular/core';
+import { GermplasmPedigreeService } from '../../shared/germplasm/service/germplasm.pedigree.service';
+import { TreeNode as PrimeNgTreeNode } from 'primeng/components/common/treenode';
+import { GermplasmNeighborhoodTreeNode } from '../../shared/germplasm/model/germplasm-neighborhood-tree-node.model';
+import { GermplasmDetailsUrlService } from '../../shared/germplasm/service/germplasm-details.url.service';
+
+@Component({
+    selector: 'jhi-germplasm-neighborhood-tree',
+    templateUrl: './germplasm-neighborhood-tree.component.html',
+    styleUrls: ['./germplasm-neighborhood-tree.component.css'],
+    encapsulation: ViewEncapsulation.None
+})
+export class GermplasmNeighborhoodTreeComponent implements OnInit {
```

Comment:
Consider making `GermplasmNeighborhoodTreeComponent` an abstract class with abstract method for returning `neighborhoodTreeNode` which would be processed by `loadTree`. 
Then having `DerivativeNeighborhoodComponent` and `MaintenanceNeighborhoodComponent` extending from this class